File: Array/#697. Degree of an Array.cpp
class Solution {
public:
    int findShortestSubArray(vector<int>& nums) {
     map<int,int>freq;
	map<int,int>pos;
	int res=0; int degree=0;
	for(int i=0; i<nums.size(); i++){
		freq[nums[i]]++;
		if(!pos.count(nums[i]))
			pos[nums[i]]=i;
		
		if(freq[nums[i]]>degree){
			degree=freq[nums[i]];
			res=i-pos[nums[i]]+1;
		}
		else if(freq[nums[i]]==degree){
			res = min(res,i-pos[nums[i]]+1);
		}
	}
	return res;
    }
};

File: Array/#697. Degree of an Array.py
class Solution:
    def findShortestSubArray(self, nums: List[int]) -> int:
        freq = defaultdict(int)
        val2idx = dict()
        max_freq = 1
        min_len = inf
        
        for i in range(len(nums)):
            if nums[i] not in val2idx:
                val2idx[nums[i]] = i

            freq[nums[i]] += 1
            if freq[nums[i]] > max_freq:
                max_freq = freq[nums[i]]
                min_len = i - val2idx[nums[i]] + 1
            elif freq[nums[i]] == max_freq:
                min_len = min(min_len, i - val2idx[nums[i]] + 1)

        return min_len



        